shell/platform/android/test/io/flutter/embedding/engine/systemchannels/PlatformChannelTest.java

```diff
@@ -0,0 +1,126 @@
+        expectedDecodedRects.add(gestureRect);
+
+        MethodCall callSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.setSystemGestureExclusionRects",
+            inputRects
+        );
+
+        platformChannel.parsingMethodCallHandler.onMethodCall(callSystemGestureExclusionRects, resultsMock);
+        verify(platformMessageHandler, times(1)).setSystemGestureExclusionRects(expectedDecodedRects);
+        verify(resultsMock, times(1)).success(null);
+    }
+
+    @Test
+    public void setSystemExclusionRectsRequiresJSONArrayInput() {
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+
+        ResultsMock resultsMock = mock(ResultsMock.class);
+        String nonJsonInput = "Non-JSON";
+        MethodCall callSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.setSystemGestureExclusionRects",
+            nonJsonInput
+        );
+        platformChannel.parsingMethodCallHandler.onMethodCall(callSystemGestureExclusionRects, resultsMock);
+
+        String inputTypeError = "Input type is incorrect. Ensure that a List<Map<String, int>> is passed as the input for SystemGestureExclusionRects.setSystemGestureExclusionRects.";
+        verify(resultsMock, times(1)).error(
+            "inputTypeError",
+            inputTypeError,
+            null
+        );
+    }
+
+    @Test
+    public void setSystemExclusionRectsSendsJSONExceptionOnIncorrectDataShape() throws JSONException {
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+
+        int top = 0;
+        int right = 500;
+
+        ResultsMock resultsMock = mock(ResultsMock.class);
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("arg1", top);
+        jsonObject.put("arg2", right);
+        JSONArray inputArray = new JSONArray();
+        inputArray.put(jsonObject);
+
+        MethodCall callSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.setSystemGestureExclusionRects",
+            inputArray
+        );
+        platformChannel.parsingMethodCallHandler.onMethodCall(callSystemGestureExclusionRects, resultsMock);
+        verify(resultsMock, times(1)).error(
+            "error",
+            "JSON error: Incorrect JSON data shape. To set system gesture exclusion rects, \n" +
+            "a JSONObject with top, right, bottom and left values need to be set to int values.",
+            null
+        );
+    }
+
+    private class ResultsMock implements Result {
```

Comment:
Please make all inner classes static to avoid hidden references that can cause leaks and complications.